CMakeLists.txt

```diff
@@ -1,24 +1,45 @@
 cmake_minimum_required(VERSION 3.1)
 set(CMAKE_CXX_STANDARD 14)
 
-project(client)
+project(client LANGUAGES CXX C VERSION 2.7.0)
+include(FeatureSummary)
+find_package(ECM 5.50.0  NO_MODULE)
+set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://projects.kde.org/projects/kdesupport/extra-cmake-modules")
+feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
+
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${ECM_MODULE_PATH})
+
+include(KDEInstallDirs)
+include(KDECMakeSettings)
+include(ECMMarkNonGuiExecutable)
+
+include(KDECompilerSettings NO_POLICY_SCOPE)
+
+set(POSITION_INDEPENDENT_CODE ON)
+set(CMAKE_AUTOUIC ON)
```

Comment:
It looks to me like our code behaves quite well with this setting so I didn't see a reason to try it on a target base.

Reply:
It enables autouic for non-Qt-targets. It's not a big deal, but it's not nice either. Especially because it's a new addition while *cleaning* *up* our CMake code.